core/ecschema-metadata/src/test/SchemaFormatsProvider.test.ts

```diff
@@ -0,0 +1,116 @@
+    const schemaXml = fs.readFileSync(schemaFile, "utf-8");
+    deserializeXmlSync(schemaXml, context);
+
+    const aecSchemaFile = path.join(__dirname, "..", "..", "..", "node_modules", "@bentley", "aec-units-schema", "AecUnits.ecschema.xml");
+    const aecSchemaXml = fs.readFileSync(aecSchemaFile, "utf-8");
+    deserializeXmlSync(aecSchemaXml, context);
+
+
+
+  });
+
+  beforeEach(() => {
+    formatsProvider = new SchemaFormatsProvider(context, "metric");
+    unitsProvider = new SchemaUnitProvider(context, UNIT_EXTRA_DATA);
+  });
+
+  it("should throw an error when format doesn't follow valid name convention", async () => {
+    await expect(formatsProvider.getFormat("nonExistentFormat")).to.be.rejected;
+  });
+  it("should return undefined when format is not found", async () => {
+    const format = await formatsProvider.getFormat("FakeSchema.nonExistentFormat");
+    expect(format).to.be.undefined;
+  });
+
+  it("should return a format from the Formats schema", async () => {
+    const format = await formatsProvider.getFormat("Formats.AmerI");
+    expect(format).not.to.be.undefined;
+  });
+
+  it("should return a format given a KindOfQuantity", async () => {
+    const format = await formatsProvider.getFormat("AecUnits.LENGTH");
+    expect(format).not.to.be.undefined;
+    expect(format?.composite?.units.length).to.be.greaterThan(0);
+  });
+
+  it("should format a length quantity to meters given a KoQ and unit provider", async () => {
+    const formatProps = await formatsProvider.getFormat("AecUnits.LENGTH");
+    expect(formatProps).not.to.be.undefined;
+    const format = await Format.createFromJSON("testFormat", unitsProvider, formatProps!);
+    const persistenceUnit = await unitsProvider.findUnitByName("Units.M"); // or unitsProvider.findUnit("m");
+    const formatSpec = await FormatterSpec.create("TestSpec", format, unitsProvider, persistenceUnit);
+
+    const result = formatSpec.applyFormatting(50);
+    expect(result).to.equal("50.0 m");
+
+  });
+
+  it("should format a length quantity to feet given a KoQ, unit provider and the imperial unit system", async () => {
+    const spy = Sinon.spy();
+    formatsProvider.onFormatsChanged.addListener(spy);
+    formatsProvider.unitSystem = "imperial";
+
+    expect(spy.calledWith({ formatsChanged: [] })).to.be.true;
+    const formatProps = await formatsProvider.getFormat("AecUnits.LENGTH_LONG");
+    expect(formatProps).not.to.be.undefined;
+    const format = await Format.createFromJSON("testFormat", unitsProvider, formatProps!);
+    const persistenceUnit = await unitsProvider.findUnitByName("Units.M"); // or unitsProvider.findUnit("m");
+    const formatSpec = await FormatterSpec.create("TestSpec", format, unitsProvider, persistenceUnit);
+
+    const result = formatSpec.applyFormatting(50);
+    expect(result).to.equal("164'0 1/2\"");
+  });
```

Comment:
Doesn't look like these tests belong here. You should be testing `SchemaFormatsProvider` rather than `FormatterSpec` and how it formats values.

Reply:
yeah these tests were more integration focused, but they aren't needed anymore now that example-code-snippets have the same test cases. Removed these tests